pkg/webhook/validatingwebhook/validate_spacebindingrequest_test.go

```diff
@@ -0,0 +1,251 @@
+				Username:         "jane",
+				MasterUserRecord: "john",
+				SpaceRole:        "maintainer", // update the SpaceRole field
+			})
+
+			// when
+			response := v.validate(req)
+
+			// then
+			// it should be allowed to update SpaceRole field
+			test.VerifyRequestAllowed(t, response, requestUID)
+		})
+	})
+
+	t.Run("creating a new SpaceBindingRequest is always allowed", func(t *testing.T) {
+		// given
+		cl := fake.NewClientBuilder().WithScheme(s).Build()
+		v := newSpaceBindingRequestValidator(cl)
+		ts := httptest.NewServer(http.HandlerFunc(v.HandleValidate))
+		defer ts.Close()
+
+		// given
+		req := newCreateSBRAdmissionRequest(t, sbrTemplate{
+			RequestUID:       requestUID,
+			Operation:        "CREATE",
+			Username:         "jane",
+			MasterUserRecord: "newMur",
+			SpaceRole:        "admin",
+		})
+
+		// when
+		response := v.validate(req)
+
+		// then
+		// it should not be allowed to create new SBR
+		test.VerifyRequestAllowed(t, response, requestUID)
+	})
+}
+
+func TestValidateSpaceBindingRequestFailsOnInvalidJson(t *testing.T) {
+	// given
+	v := &SpaceBindingRequestValidator{}
+
+	t.Run("object is not spacebindingrequest", func(t *testing.T) {
+		// when
+		response := v.validate(test.IncorrectRequestObjectJSON)
+
+		// then
+		test.VerifyRequestBlocked(t, response, "unable to unmarshal object or object is not a spacebindingrequest", "a68769e5-d817-4617-bec5-90efa2bad6f8")
+	})
+
+	t.Run("json is invalid", func(t *testing.T) {
+		// when
+		rawJSON := []byte(`something wrong !`)
+		response := v.validate(rawJSON)
+
+		// then
+		test.VerifyRequestBlocked(t, response, "cannot unmarshal string into Go value of type struct", "")
+	})
+}
+
+func TestValidateSpaceBindingRequestFailsOnGettingSBR(t *testing.T) {
+	// given
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	cl := commontest.NewFakeClient(t)
+	cl.MockGet = mockGetSpaceBindingRequestFail(cl)
+	v := newSpaceBindingRequestValidator(cl)
+	ts := httptest.NewServer(http.HandlerFunc(v.HandleValidate))
+	defer ts.Close()
+	req := newCreateSBRAdmissionRequest(t, sbrTemplate{
+		RequestUID:       "xvadsfasdf",
+		Operation:        "UPDATE",
+		Username:         "jane",
+		MasterUserRecord: "newMur",
+		SpaceRole:        "admin",
+	})
+
+	// when
+	response := v.validate(req)
+
+	// then
+	test.VerifyRequestBlocked(t, response, "unable to check if spacebindingrequest already exists. SpaceBindingRequest.Name: john-sbr: mock error", "xvadsfasdf")
+}
+
+func newSpaceBindingRequestValidator(cl runtimeClient.Client) *SpaceBindingRequestValidator {
+	return &SpaceBindingRequestValidator{
+		Client: cl,
+	}
+}
+
+func newSBR(name, namespace, mur, spaceRole string) *toolchainv1alpha1.SpaceBindingRequest {
+	return &toolchainv1alpha1.SpaceBindingRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				toolchainv1alpha1.ProviderLabelKey: toolchainv1alpha1.ProviderLabelValue,
+			},
```

Comment:
why do we need this label set?

Reply:
for the sake of the test we don't , was a copy paste left over I guess. Removed in https://github.com/codeready-toolchain/member-operator/pull/470/commits/9ddde444e5b80865e6c6793321a2dfcfff9f15b0